client/client.go

```diff
@@ -2144,10 +2153,32 @@ func (c *Client) allocSync() {
 			if len(updates) == 0 {
 				continue
 			}
+			// Ensure we never send an update before we've had at least one sync
+			// from the server
+			select {
+			case <-c.serversContactedCh:
+			default:
+				continue
+			}
 
 			sync := make([]*structs.Allocation, 0, len(updates))
-			for _, alloc := range updates {
-				sync = append(sync, alloc)
+			for allocID, update := range updates {
+				if ar, ok := c.allocs[allocID]; ok {
```

Comment:
c.allocs needs to hold `allocLocks` right?

Reply:
Oof, yes. Will fix.

Reply:
Resolved in f62c33f7e4fc40db35ed3287d8089e6258a3fe58